### PR TITLE
JDK-8283670: gtest os.release_multi_mappings_vm is still racy

### PR DESCRIPTION
**Pull request by tstuefe**
[JDK-8280940](https://bugs.openjdk.java.net/browse/JDK-8280940) fixed up os.release_multi_mappings_vm to make its handling of multiple neighboring reservations less racy, but I forgot to fix up the reserve_multiple() function, which follows the same pattern and is still racy. It reserves an area, releases it, then re-reserves stripes in that area.

We see intermittent problems on Linux PPC. This makes sense because we have here 64k pages, and therefore the stripe size is large, and we have a higher chance of someone grabbing that address space concurrently between the release and the stripe-wise re-reserve.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283670](https://bugs.openjdk.java.net/browse/JDK-8283670): gtest os.release_multi_mappings_vm is still racy

### Reviewers
 * [Martin Doerr](https://openjdk.java.net/census#mdoerr) (@TheRealMDoerr - **Reviewer**) ⚠️ Review applies to 669cb5d37f5225382edeecbb757859516923017a
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 669cb5d37f5225382edeecbb757859516923017a

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7953/head:pull/7953` \
`$ git checkout pull/7953`

Update a local copy of the PR: \
`$ git checkout pull/7953` \
`$ git pull https://git.openjdk.java.net/jdk pull/7953/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7953`

View PR using the GUI difftool: \
`$ git pr show -t 7953`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7953.diff">https://git.openjdk.java.net/jdk/pull/7953.diff</a>

</details>

### CODE REVIEW

**Review by TheRealMDoerr — APPROVED** [review]
LGTM.

**Comment by tstuefe** [comment]
> LGTM.

Thank you Martin!

Could I have a second review to get this on its way?

File: test/hotspot/gtest/runtime/test_os.cpp
Comment (dholmes-ora) [inline test/hotspot/gtest/runtime/test_os.cpp:419]: Nit: you don't need to check this as `carefully_release_multiple` is a no-op for `num_stripes==0`.

**Review by dholmes-ora — APPROVED** [review]
Looks good. One minor nit below.

**Comment by tstuefe** [comment]
Thanks @dholmes-ora.

/integrate

**Pull request closed without merging**
